### PR TITLE
[icu] Fix osx release only builds

### PR DESCRIPTION
**Pull request by m-kuhn**
We don't necessarily build "debug", in case it's not built, avoid looping over this config in https://github.com/microsoft/vcpkg/pull/34539/files#diff-ea2b03d0dcb43b39cfac7b4f96e0ba52a5e4f15c1df6986850e0d52d56229430R94

### CODE REVIEW

File: ports/icu/portfile.cmake [hunk 1]
```diff
@@ -34,7 +34,9 @@ list(APPEND CONFIGURE_OPTIONS_RELEASE --disable-debug --enable-release)
 list(APPEND CONFIGURE_OPTIONS_DEBUG  --enable-debug --disable-release)
 
 set(RELEASE_TRIPLET ${TARGET_TRIPLET}-rel)
-set(DEBUG_TRIPLET ${TARGET_TRIPLET}-dbg)
+if (NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
+  set(DEBUG_TRIPLET ${TARGET_TRIPLET}-dbg)
+endif()
```
Comment (dg0yt) [inline ports/icu/portfile.cmake:39]: Setting the variable as before doesn't really break anything, does it?

Comment (m-kuhn) [reply ports/icu/portfile.cmake:39]: If the variable is set, we will loop over it here and try to fixup rpath on an inexistent file. It's not very self explanatory ... We'd probably better have one list to loop over than two separate variables.

https://github.com/microsoft/vcpkg/pull/34539/files#diff-ea2b03d0dcb43b39cfac7b4f96e0ba52a5e4f15c1df6986850e0d52d56229430R94

Comment (m-kuhn) [reply ports/icu/portfile.cmake]: New version pushed

**Comment by jimwang118** [comment]
Compile test pass with following triplets:
```
x64-osx
```

**Comment by data-queue** [comment]
I think I'm missing some context. Can you explain what these changes do?

**Comment by m-kuhn** [comment]
If we only build `-rel` but do not build `-dbg`, we currently get a build failure, as this port tries to access (inexistant) files of the `-dbg` build in the abovementioned loop.

**Pull request merged**